refocus-bdk-server.js

```diff
@@ -188,46 +204,87 @@ module.exports = (config) => {
 
     const socket = io.connect(connectUrl, opts);
 
-    socket.on(initalizeEventName, (data) => {
+    socket.on(initalizeEventName, async (data) => {
       const eventData = JSON.parse(data);
       const room = eventData[initalizeEventName];
+      const { id, updatedAt } = room;
+      const hasAlreadyBeenConsumed = cache &&
+        await cache.hasBeenConsumed(id, updatedAt, initalizeEventName);
+      if (hasAlreadyBeenConsumed) {
+        return;
+      }
       app.emit('refocus.internal.realtime.bot.namespace.initialize', room);
-      log.realtime('New Room', room);
     });
 
-    socket.on(settingsChangedEventName, (data) => {
+    socket.on(settingsChangedEventName, async (data) => {
       const eventData = JSON.parse(data);
       const room = eventData[settingsChangedEventName];
+      const { id, updatedAt } = room;
```

Comment:
is this a room ID or event ID?

Reply:
good point, it's a roomID - makes me think that if multiple bots use the same redis and are responding to the same event then we'll have trouble. Think I'll need to include bot name in key as well

Reply:
yes, bot name solves the problem, I thought the event would have its own id... something like room.event.id